### PR TITLE
feat: Acquire configuration from remote, when sampler is started 

### PR DESCRIPTION
**Pull request by tttoad**
feat: load the configuration from the remote as soon as the sampler is started and the type is remote, instead of waiting for the timer notification

### CODE REVIEW

**Review by dmathieu — COMMENTED** [review]
While this change in itself makes sense, it would be nice if it were tested as well.

**Comment by tttoad** [comment]
> While this change in itself makes sense, it would be nice if it were tested as well.
I'm doing local testing, which will take a little time since jaeger-remote is not compatible with the old interface.

**Comment by tttoad** [comment]
> I meant adapting the unit tests to check this behavior change automatically.
Okay, I'll add the unit tests. I'll probably be able to submit it tomorrow or the next day.

**Comment by tttoad** [comment]
> I meant adapting the unit tests to check this behavior change automatically.

I have added the unit test,PTAL.

File: samplers/jaegerremote/sampler_remote_test.go [hunk 1]
```diff
@@ -502,3 +508,26 @@ func getSamplingStrategyResponse(strategyType jaeger_api_v2.SamplingStrategyType
 	}
 	return nil
 }
+
+func TestRemotelyControlledSampler_ImmediatelyUpdateOnStartup(t *testing.T) {
+	initSampler := newProbabilisticSampler(0.123)
+	fetcher := &testSamplingStrategyFetcher{response: []byte("rateLimiting")}
```
[the remark this answers is not in the document]
Comment (tttoad) [reply samplers/jaegerremote/sampler_remote_test.go:514]: The `probabilistic` strategy is  default strategy when reading strategy from remote fails.

File: samplers/jaegerremote/sampler_remote_test.go [hunk 2]
```diff
@@ -502,3 +508,26 @@ func getSamplingStrategyResponse(strategyType jaeger_api_v2.SamplingStrategyType
 	}
 	return nil
 }
+
+func TestRemotelyControlledSampler_ImmediatelyUpdateOnStartup(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (tttoad) [reply samplers/jaegerremote/sampler_remote_test.go:512]: You are right.

File: samplers/jaegerremote/sampler_remote_test.go [hunk 3]
```diff
@@ -281,6 +287,29 @@ func TestRemotelyControlledSampler_updateSampler(t *testing.T) {
...
+		WithSamplingRefreshInterval(10*time.Minute),
+		withSamplingStrategyFetcher(fetcher),
+		withSamplingStrategyParser(parser),
+		withUpdaters(updaters...),
+	)
+	time.Sleep(100 * time.Millisecond) // waiting for s.pollController
+	sampler.Close()                    // stop pollController, avoid date race
+	s, ok := sampler.sampler.(*rateLimitingSampler)
+	assert.True(t, ok)
+	assert.Equal(t, s.maxTracesPerSecond, float64(100))
```
Comment (dmathieu) [inline samplers/jaegerremote/sampler_remote_test.go:310]:
```suggestion
	assert.Equal(t, float64(100), s.maxTracesPerSecond)
```

This way, if it fails, the "expected" value is the proper one.

File: samplers/jaegerremote/sampler_remote_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmathieu) [reply samplers/jaegerremote/sampler_remote_test.go]: What I mean is that all other tests specify the probabilistic strategy and check that up. So relying on that here would be more consistent.
At the same time, being able to rely on different strategies within the tests is fine. So 👍

File: samplers/jaegerremote/sampler_remote_test.go (hunk 3 above)
Comment (tttoad) [reply samplers/jaegerremote/sampler_remote_test.go]: Thanks for pointing out.  PTAL.

**Review by dmathieu — APPROVED** [review]

**Comment by tttoad** [comment]
@hanyuancheung  The CI error looks like the execution timed out? Maybe a retry will fix it.

**Review by hanyuancheung — APPROVED** [review]
Maybe re-push it with '-f' will be helpful.

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**